.github/workflows/integration-test-custom.yml

```diff
@@ -0,0 +1,260 @@
+              if nc -zvw3 ${{ steps.create_vm.outputs.publicIpAddress }} 22; then
+                echo "SSH is available!"
+                exit 0
+              fi
+              sleep 10
+            done
+            echo "SSH is not available after 2 minutes; failing"
+            exit 1
+
+      - name: Install Docker
+        uses: appleboy/ssh-action@master
+        with:
+          host: ${{ steps.create_vm.outputs.publicIpAddress }}
+          username: azureuser
+          key: ${{ secrets.SSH_PRIVATE_KEY }}
+          script: |
+            curl -fsSL https://get.docker.com -o get-docker.sh
+            sudo sh get-docker.sh
+            sudo systemctl enable docker
+            sudo systemctl start docker
+            sudo usermod -aG docker $USER
+            sudo curl -L "https://github.com/docker/compose/releases/download/v2.20.0/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+            sudo chmod +x /usr/local/bin/docker-compose
+
+      - name: Clone Repos and Setup Environment
+        uses: appleboy/ssh-action@master
+        env:
+          GITHUB_TOKEN: ${{ secrets.IT_GH_ACCESS_TOKEN }}
+          REPO_REFS: ${{ inputs.repo_refs }}
+        with:
+          host: ${{ steps.create_vm.outputs.publicIpAddress }}
+          username: azureuser
+          key: ${{ secrets.SSH_PRIVATE_KEY }}
+          envs: GITHUB_TOKEN,REPO_REFS
+          script: |
+            # Function to get ref for a repo
+            get_ref() {
+              echo "$REPO_REFS" | tr ',' '\n' | grep "^$1:" | cut -d':' -f2 || echo "master"
+            }
+            
+            # First clone the compose repo
+            COMPOSE_REPO="${{ env.COMPOSE_REPO }}"
+            COMPOSE_REF=$(get_ref "$COMPOSE_REPO")
+            echo "Cloning compose repo https://github.com/${COMPOSE_REPO}.git branch/ref: ${COMPOSE_REF}"
+            git clone "https://github.com/${COMPOSE_REPO}.git" -b "$COMPOSE_REF"
+            
+            # Navigate to compose repo
+            REPO_NAME=$(echo ${COMPOSE_REPO} | cut -d'/' -f2)
+            cd $REPO_NAME
+            
+            # Clone other repositories inside the compose repo directory
+            echo "${{ env.REPOS }}" | while read repo; do
+              if [ ! -z "$repo" ] && [ "$repo" != "$COMPOSE_REPO" ]; then
+                REF=$(get_ref "$repo")
+                echo "Cloning https://github.com/${repo}.git branch/ref: ${REF}"
+                git clone "https://github.com/${repo}.git" -b "$REF"
+              fi
+            done
+            
+            # Copy sample_env to .env and replace tokens
+            cp sample_env .env
+            
+            # Replace both GitHub tokens with the provided token
+            sed -i "s/^CRAWLER_GITHUB_TOKEN=.*/CRAWLER_GITHUB_TOKEN=${GITHUB_TOKEN}/" .env
+            sed -i "s/^CURATION_GITHUB_TOKEN=.*/CURATION_GITHUB_TOKEN=${GITHUB_TOKEN}/" .env
+            echo "" >> .env && echo "CRAWLER_SCANCODE_PARALLELISM=14" >> .env
```

Comment:
AS a separate PR, could we document this in https://github.com/clearlydefined/operations/blob/main/overview.md?